okapi/services/attrs/attribute-definitions.xml

```diff
@@ -3976,4 +4032,126 @@ for the original implementations.
         </lang>
     </attr>
 
+    <attr acode="A78" categories="de-cache-types">
+        <ocgs id="216" inc="true" />
+        <opencaching schema="OCUS" id="216" />
+        <lang id="en">
+            <name>Contains advertising</name>
+            <desc>
+                The cache contains commercial advertising and was likely placed by a business.
+            </desc>
+        </lang>
+    </attr>
+
+    <attr acode="A79" categories="de-cache-types">
+        <ocgs id="217" inc="true" />
+        <opencaching schema="OCUK" id="217" />
+        <lang id="en">
+            <name>Military training area</name>
+            <desc>
+                Military training area, some access restrictions - check before visit.
+            </desc>
+        </lang>
+    </attr>
+
+    <attr acode="A80" categories="de-cache-types">
+        <ocgs id="218" inc="true" />
+        <opencaching schema="OCRO" id="218" />
+        <lang id="en">
+            <name>Video surveillance</name>
+            <desc>
+                Area under video surveillance. Exercise caution.
+            </desc>
+        </lang>
+    </attr>
+
+    <attr acode="A81" categories="de-cache-types">
+        <ocgs id="219" inc="true" />
+        <opencaching schema="OCRO" id="219" />
+        <lang id="en">
+            <name>Trackables</name>
+            <desc>
+                Suitable to hold trackables.
+            </desc>
+        </lang>
+    </attr>
+
+    <attr acode="A82" categories="de-cache-types">
+        <groundspeak id="54" inc="true" name="Abandoned structure" />
+        <opencaching schema="OCRO" id="54" />
+        <lang id="en">
+            <name>Ruin</name>
+            <desc>
+                Abandoned structure / ruin.
+            </desc>
+        </lang>
+    </attr>
+
+    <attr acode="A83" categories="de-cache-types">
+        <groundspeak id="48" inc="true" name="UV Light Required" />
+        <opencaching schema="OCRO" id="48" />
+        <lang id="en">
+            <name>UV Light Required</name>
+            <desc>
+                UV Light is required to find clues to the cache location.
+            </desc>
+        </lang>
+    </attr>
+
+    <attr acode="A84" categories="de-cache-types">
+        <groundspeak id="15" inc="false" name="Available during winter" />
+        <opencaching schema="OCUS" id="9015" />
+        <lang id="en">
+            <name>NOT available during winter</name>
+            <desc>
+                NOT available during winter.
+            </desc>
+        </lang>
+    </attr>
+
+    <attr acode="A85" categories="de-cache-types">
+        <groundspeak id="1" inc="false" name="Dogs" />
```

Comment:
Should GS ID 1 (dogs) really map to A85 (no dogs)?
This is also not so described in the wiki entry.

Reply:
Ok, this is a negative ID. 
But then your wiki page had be to changed for ID 9013 (remove gc_id 13).

Reply:
A85 corresponds to GC id=1 with inc=0 (no dogs allowed, negative meaning)

Reply:
Wiki page is correct regarding A40, the internal implementation at OCDE has separate columns regarding `gc_id` and `inc`.

Reply:
Not to me.

Please compare the state of ID 14/9014 with ID 13/9013 in the Wiki table and attribute-definitions.txt. 
In the file attribute-definitions.txt the characters 41 and 42 have both entries <groundspeak id="14".....
and acode 39 and 40 have <groundspeak id="13" .....

So I think that the documentation in the wiki should only be implemented with one kind and not with two different ones.

Reply:
```
    <attr acode="A40" categories="de-time-and-seasons">
        <groundspeak id="13" inc="false" name="Available at all times" />
        <opencaching schema="OCDE" id="39" />
        <opencaching schema="OCNL" id="9013" />
        <opencaching schema="OCRO" id="9013" />
        <opencaching schema="OCUK" id="9013" />
        <opencaching schema="OCUS" id="9013" />
```
Wiki:
`9013 | NOT available 24/7 | Available only during open hours | 39 | 13 | time | Only available at specified times | 0 | A40`

-----------------------------
Then: 
```
    <attr acode="A41" categories="de-time-and-seasons">
        <groundspeak id="14" inc="false" name="Recommended at night" />
        <opencaching schema="OCDE" id="40" />
        <opencaching schema="OCUK" id="9014" />
```
Wiki:
`9014 | Day | NOT recommended at night | 40 | 14 | day | By day only | 0 | A41 `

-----------------------------
Then: 
```
    <attr acode="A42" categories="de-time-and-seasons">
        <groundspeak id="14" inc="true" name="Recommended at night" />
        <opencaching schema="OCPL" id="14" />
        <opencaching schema="OCNL" id="14" />
        <opencaching schema="OCRO" id="14" />
```

Wiki:
`14 | Night | Recommended at night |   |   |   |   |   | A42`

-----------------------------

Which is correct as intended. 
- **A39** is Groundspeak attr 13; **inc=true** (Available 24/7)
- **A40** is Groundspeak attr 13; **inc=false** (NOT available 24/7)
 with OCDCE data being: internal ID=39, gc_id=13, inc=0 (false)
and OCPL data being: internal ID=9013.

OC attributes are not 3-state like GC attributes. Negating an attribute is represented by a standalone attribute whose meaning is the negation of another attribute. 
OKAPI does it's best to include compatibility information such that each OC attribute (where possible) may be represented by it's Groundspeak equivalent.
With this in mind, it is perfectly normal for A39 and A40 to reference the same Groundspeak attribute ID (13 in this case).
Same logic applies to A41 and A42.

Furthermore, since OCDE and OCPL implementations differ, OKAPI also does it's best to present a unified API and a unified view. 
For all intents and purposes, third party apps using OKAPI should use the ACODES to represent attributes. Using the `<groundspeak>` tag being for compatibility with existing app code and data to handle Groundspeak cache information.  All the rest are **internals** should the app need or desire to present such detailed differences. 

The logic is explained in the section [Attribute_numbering_rules](https://wiki.opencaching.eu/index.php?title=Cache_attributes#Attribute_numbering_rules) from the Wiki.